### ISSUE TITLE
Leap Second test failure on 4.1rc1 (expired builtin file?)

### ISSUE THREAD

**Issue by eteq** [astropy/astropy#10403]
When testing Astropy v4.1rc1 on Windows 10/WSL2 + Ubuntu 18.04 (if you don't know WSL, it's basically "windows inside linux"), I see the following test failure in Astropy v4.1rc1:
```
_______________________________ TestDefaultAutoOpen.test_system_file_used_if_not_expired _______________________________

self = <astropy.utils.iers.tests.test_leap_second.TestDefaultAutoOpen object at 0x7f434d88bad0>
tmpdir = local('/tmp/pytest-of-erik/pytest-6/test_system_file_used_if_not_e0')

    @pytest.mark.skipif(not os.path.isfile(SYSTEM_FILE),
                        reason=f'system does not have {SYSTEM_FILE}')
    def test_system_file_used_if_not_expired(self, tmpdir):
        # We skip the test if the system file is on a CI and is expired -
        # we should not depend on CI keeping it up to date, but if it is,
        # we should check that it is used if possible.
        if (iers.LeapSeconds.open(SYSTEM_FILE).expires <= self.good_enough
                and ON_CI):
            pytest.skip("CI system leap second file is expired.")

        self.remove_auto_open_files('erfa')
        with iers.conf.set_temp('system_leap_second_file', SYSTEM_FILE):
            ls = iers.LeapSeconds.open()
            assert ls.expires > self.good_enough
            assert ls.meta['data_url'] in (iers.IERS_LEAP_SECOND_FILE,
                                           SYSTEM_FILE)

            # Also check with a "built-in" file that is expired
            fake_file = make_fake_file('28 June 2017', tmpdir)
            iers.LeapSeconds._auto_open_files[0] = fake_file
            ls2 = iers.LeapSeconds.open()
            assert ls2.expires > Time.now()
>           assert ls2.meta['data_url'] == SYSTEM_FILE
E           AssertionError: assert 'https://hpie...ap_Second.dat' == '/usr/share/z...-seconds.list'
E             - /usr/share/zoneinfo/leap-seconds.list
E             + https://hpiers.obspm.fr/iers/bul/bulc/Leap_Second.dat

miniconda3/envs/test-ap41/lib/python3.7/site-packages/astropy/utils/iers/tests/test_leap_second.py:269: AssertionError
```
the `/usr/share/zoneinfo/leap-seconds.list` file on this setup ends at 37 (the one at the end of 2016), which is up-to-date...

The test itself actually confuses me though - I don't understand why the `SYSTEM_FILE` would be the same as the `fake_file`... Perhaps it's that this Leap_Second.dat expired at a time not expected by this test? (@mhvk may have some insight of the intent of the test here...)

**Comment by eteq** [comment]
It's not clear to me this is actually a bug in the code, btw - it may be a flaw in the test assumptions.

**Comment by mhvk** [comment]
I think I may finally just have to bite the bullet and accept that this test should be removed, as @pllim set us up to do a whole long while back in #9720.

The idea with the test is that if a system file is available which is better than the one bundled with astropy, it should be possible to set it up so that it is automatically used. And on any linux system at least, one should be able to count on the system leap-seconds file to be up to date. But I fear that, as on CI, in your "inside windows" setup, the leap-second file is in fact expired, and therefore it downloads a more recent one. Which, arguably, is exactly what the code should do... 

So, perhaps the conclusion is that you are right, that the test is written wrong: it should be executed only if a system file is present *and* if the file is not expired. I.e., maybe just remove the `and ON_CI` clause in the `if` statement near the start... I should not really try to aim to get people to keep their systems up to date by getting a failing astropy test!

**Comment by asreimer** [comment]
I'm not sure how helpful this is, but I thought I would report that I'm seeing this test fail inside an Ubuntu 20.04 docker container on python 3.8 with astropy 4.1rc1. The assertion error for me looks like:

    AssertionError: assert 'https://hpiers.obspm.fr/iers/bul/bulc/Leap_Second.dat' in ('/home/jovyan/envs/py38/lib/python3.8/site-packages/astropy/utils/iers/data/Leap_Second.dat', '/usr/share/zoneinfo/leap-seconds.list')

**Comment by pllim** [comment]
@asreimer , did you try 4.1rc2?

**Comment by taldcroft** [comment]
@pllim - I wasn't aware that rc2 is out. Did you mean rc1?

**Comment by pllim** [comment]
Eh... I swear I saw 4.1rc2 on https://pypi.org/project/astropy/#history a few weeks ago. Some people even tested it (see https://github.com/astropy/astropy/wiki/astropy-v4.1-release-candidate-(RC)-Testing#rc2). Where did it go?

**Comment by asreimer** [comment]
@pllim Just tried it now (`pip install git+https://github.com/astropy/astropy@v4.1rc2`). As expected from #9720 being merged, the problem is gone.

I didn't expect my original comment to be that useful, except that it might add to the weight of the decision to remove the test (it fails on multiple OSes in different versions of Python, so it's a problem specifically with the test).

**Comment by pllim** [comment]
Thanks, @asreimer ! Great idea to look at the tags and thanks for proving that I am not crazy here. I guess this is effectively fixed then for 4.1 and can be closed. :)

**Comment by eteq** [comment]
To close the loop above for posterity: rc2 was "soft released" in that the tag was created before we realized there was a problem with the wheels-building machinery.  So that's why there was a window where it existed but not everything was announced.
